src/functions/functionalPolicyFactory.ts

```diff
@@ -0,0 +1,55 @@
+import { APIGatewayAuthorizerResult, Statement } from "aws-lambda";
+import newPolicyDocument from "./newPolicyDocument";
+import { ILogEvent } from "../models/ILogEvent";
+import { HttpVerb } from "../services/http-verbs";
+import StatementBuilder from "../services/StatementBuilder";
+
+type NonEmptyArray<T> = [T, ...T[]];
+
+interface IApiAccess {
+  verb: HttpVerb;
+  path: string;
+}
+
+const functionConfig: { [key: string]: NonEmptyArray<IApiAccess> } = {
+  "TechRecord.Amend": [
+    {
+      verb: "POST",
+      path: "vehicles/*",
+    },
+    {
+      verb: "PUT",
+      path: "vehicles/*",
+    },
+  ],
+  "TechRecord.View": [
+    {
+      verb: "GET",
+      path: "vehicles/*",
+    },
+  ],
+};
```

Comment:
2 Things:
1. Is this the best place to maintain this list of VTM specific functional roles? (a separate config only file might be easier to discover perhaps)

2. There are some `TestResult.*` permissions that we might wish to include in this initial implementation:
```suggestion
const functionConfig: { [key: string]: NonEmptyArray<IApiAccess> } = {
  "TechRecord.Amend": [
    {
      verb: "POST",
      path: "vehicles/*",
    },
    {
      verb: "PUT",
      path: "vehicles/*",
    },
  ],
  "TechRecord.View": [
    {
      verb: "GET",
      path: "vehicles/*",
    },
  ],
  "TestResult.Amend": [
    {
      verb: "POST",
      path: "test-result/*",
    },
    {
      verb: "PUT",
      path: "test-result/*",
    },
    {
      verb: "GET",
      path: "test-types/*",
    },
    {
      verb: "GET",
      path: "test-stations/*",
    },
    {
      verb: "GET",
      path: "defects/*",
    },
    {
      verb: "GET",
      path: "reference/*",
    },
  ],
  "TestResult.View": [
    {
      verb: "GET",
      path: "test-result/*",
    },
    {
      verb: "GET",
      path: "test-types/*",
    },
    {
      verb: "GET",
      path: "v1/document-retrieval/*",
    },
};
```

Reply:
knew this wasn't the full list so I really appreciate you helping me out with that! 👍 
Can't see a downside to exporting the config into its own file - I'll get that done next.